app/services/notification_service.rb

```diff
@@ -7,32 +7,163 @@ def self.renderer
     )
   end
 
+  def self.get_template_for_event_type(event_type)
+    'map_mailer/' + event_type
+  end
+  
+  def self.get_mailboxer_code_for_event_type(event_type)
+    case event_type
+      when 'access_approved'
+        MAILBOXER_CODE_ACCESS_APPROVED
+      when 'access_request'
+        MAILBOXER_CODE_ACCESS_REQUEST
+      when 'invite_to_edit'
+        MAILBOXER_CODE_INVITE_TO_EDIT
+
+      when 'map_activity'
+        MAILBOXER_CODE_MAP_ACTIVITY
+      when 'map_collaborator_added'
+        MAILBOXER_CODE_MAP_COLLABORATOR_ADDED
+      when 'map_updated'
+        MAILBOXER_CODE_MAP_UPDATED
+      when 'map_message'
+        MAILBOXER_CODE_MAP_MESSAGE
+      when 'map_starred'
+        MAILBOXER_CODE_MAP_STARRED
+        
+      when 'topic_added_to_map'
+        MAILBOXER_CODE_TOPIC_ADDED_TO_MAP
+      when 'topic_connected'
+        MAILBOXER_CODE_TOPIC_CONNECTED
+      when 'topic_deleted'
+        MAILBOXER_CODE_TOPIC_DELETED
+      when 'topic_disconnected'
+        MAILBOXER_CODE_TOPIC_DISCONNECTED
+      when 'topic_updated'
+        MAILBOXER_CODE_TOPIC_UPDATED
+    end
+  end
+
+  def self.get_subject_for_event(entity, event_type, event)
+    case event_type
+      when 'access_approved' # event is an AccessRequest
+        entity.name + ' - access approved' 
+      when 'access_request' # event is an AccessRequest
+        entity.name + ' - request to edit'
+      when 'invite_to_edit' # event is a UserMap
+        entity.name + ' - invited to edit'
+
+      when 'map_activity' #disabled
+        'placeholder'
+      when 'map_collaborator_added' #disabled
+        'placeholder'
+      when 'map_updated' #disabled
+        'placeholder'
+      when 'map_message' # event is a Message
+        entity.name + ' - received a chat message'
+      when 'map_starred' # event is a Star
+        entity.name + ' was starred by ' + event.user.name
+        
+      when 'topic_added_to_map' # event is an Event::TopicAddedToMap
+        entity.name + ' was added to map ' + event.map.name
+      when 'topic_connected' # event is a Synapse
+        'new synapse to topic ' + entity.name
+      when 'topic_deleted' #disabled
+        'placeholder'
+      when 'topic_disconnected' #disabled
+        'placeholder'
+      when 'topic_updated' #disabled
+        'placeholder'
+    end
+  end
```

Comment:
this might be overkill. can't this go in individual mailers? if not, whateva, but it seems too centralized to get the subject here